support-frontend/assets/helpers/forms/paymentIntegrations/oneOffContributions.ts

```diff
@@ -147,7 +148,10 @@ function paymentResultFromObject(
 		});
 	}
 
-	return Promise.resolve(PaymentSuccess);
+	const response = json as { data?: { userType: UserType } };
```

Comment:
Admittedly a bit of a bodge 🤷‍♂️

Reply:
Would it be worth writing a zod schema to validate this, rather than casting?

Reply:
Yeah that would be ideal, there's another cast in this function so definitely a worthwhile refactor